.eslintrc.yml

```diff
@@ -0,0 +1,54 @@
+---
+  parserOptions:
+    ecmaVersion: 6
+  env:
+    browser: true
+    jquery: true
+  extends: eslint:recommended
+  rules:
+    no-console: off
+    no-unused-vars:
+     - error
+     - vars: local
+       args: none
+    no-empty: off
+    no-shadow: error
+    no-mixed-spaces-and-tabs: [error, smart-tabs]
+    id-length: [error, exceptions: [i, j, e, x, y]]
+
+    indent: [error, tab]
+    semi: error
+    brace-style: [error, 1tbs, allowSingleLine: true]
+    linebreak-style: [error, unix]
+    semi-spacing: error
+    comma-spacing: error
+    space-infix-ops: error
+    space-before-function-paren: [error, never]
+    space-before-blocks: error
+    keyword-spacing:
+      - error
+      - before: true
+        after: true
+  globals:
+    require: true
+    unsafeWindow: true
+
+    XKit: true
+    XBridge: true
+    Tumblr: true
+    add_tag: true
```

Comment:
Tumblr and add_tag being here are a compromise between explicitness and usability. I would prefer that we would have to inline declare them as globals in every function that gets injected into the browser scope, but a) this is somewhat impractical and b) the way eslint directives work the protection provided by doing it the other way is minimal